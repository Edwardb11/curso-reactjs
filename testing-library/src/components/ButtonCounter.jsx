import React from 'react'

function ButtonCounter({name,value=1, action}) {
    
  return (
    <button aria-label={name} onClick={() => action(value)}>
    {value}
  </button>
  )
}

export default ButtonCounter