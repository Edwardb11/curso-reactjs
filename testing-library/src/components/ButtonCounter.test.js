import { render, screen } from "@testing-library/react";
import ButtonCounter from "./ButtonCounter";

describe("<ButtonCounter/>", () => {
  // simulacion => Mock scaneo
  const action = jest.fn();

  beforeEach(() => {
    render(<ButtonCounter name="aumentar" value={3} action={action} />);
  });
  it("Cree un boton correctamente", () => {
    // Espero que esta validacion exista en el documento
    expect(screen.getByLabelText("aumentar")).toBeInTheDocument();
  });
  // it("Cree un boton correctamente sin mandar value", () => {
  //   render(<ButtonCounter name="aumentar"  action={action} />);
  //   expect(screen.getByLabelText("aumentar")).toBeInTheDocument();
  // });

  it("Cree un boton correctamente  Snapshot", () => {
    // Espero que esta validacion exista en el documento
    expect(screen.getByLabelText("aumentar")).toMatchSnapshot();
  });
});
