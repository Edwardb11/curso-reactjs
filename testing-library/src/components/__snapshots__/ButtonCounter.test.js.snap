// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ButtonCounter/> Cree un boton correctamente  Snapshot 1`] = `
<button
  aria-label="aumentar"
>
  3
</button>
`;
